h/static/scripts/util/search-text-parser.js

```diff
@@ -122,7 +116,73 @@ function getLozengeValues(queryString) {
   };
 }
 
+/**
+ * Return true if the input query term string has a known named
+ * query term.
+ *
+ * @param {String} queryTerm The query term string.
+ *
+ * @returns {Bool} True if the query term string has a known named
+ * query term and False otherwise.
+ *
+ * @example
+ * // returns false
+ * hasKnownNamedQueryTerm('foo:bar')
+ * @example
+ * // returns true
+ * hasKnownNamedQueryTerm('user:foo')
+ */
+function hasKnownNamedQueryTerm(queryTerm) {
+  const knownNamedQueryTerms = ['user',
+    'url',
+    'group',
+    'tag'];
+
+  var facetName;
+
+  facetName = getLozengeFacetNameAndValue(queryTerm).facetName;
+  if (knownNamedQueryTerms.indexOf(facetName) >= 0) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Return an object with the facet name and value for a given query term.
+ *
+ * @param {String} queryTerm The query term string.
+ *
+ * @returns {Object} An object with two properties:
```

Comment:
The return type doesn't match the description if the query term does not contain a facet separator. There are several ways you could address this. One would be to change the return type to `null|Object` where the result is null if there is no facet. The other would be to always return an object with `facetName` and `facetValue` properties but make `facetName` an empty string if there is no separator. My inclination would be to use the latter style as this will probably result in fewer paths through the code.